mbf_costmap_nav/src/mbf_costmap_nav/costmap_navigation_server.cpp

```diff
@@ -885,7 +886,26 @@ bool CostmapNavigationServer::callServiceFindValidPose(mbf_msgs::FindValidPose::
   response.pose.pose.position.x = sol.pose.x;
   response.pose.pose.position.y = sol.pose.y;
   response.pose.pose.position.z = 0;
-  response.pose.pose.orientation = tf::createQuaternionMsgFromYaw(sol.pose.theta);
+
+  // if the difference between solution angle and requested angle is less than ANGLE_INCREMENT,
+  // use the requested one to avoid violating a very small angle_tolerance (e.g. 0)
+  response.pose.pose.orientation =
+      std::abs(angles::shortest_angular_distance(goal.theta, sol.pose.theta)) < ANGLE_INCREMENT ?
```

Comment:
So since `ANGLE_INCREMENT` is now max step size, this check isn't valid anymore since the search will make smaller steps internally.

Not sure how to best accommodate this. Maybe we could add a min_step_size as well, and use that here? Or return more information from `free_pose_search.search();` (e.g. by how much the goal was displaced)?

Reply:
yes, the check is invalid now. 
The information of how much the goal was displaced is available to the user since he has the goal and he has the result. I think we can just check if the input == output (after conversion) and, in this case, use the requested angle (without conversion), avoiding floating erros.